src/main/java/org/gaul/s3proxy/S3ProxyHandler.java

```diff
@@ -1610,6 +1655,11 @@ private void sendSimpleErrorResponse(HttpServletResponse response,
         }
     }
 
+    public void setProviders(
+            Map<String, Map.Entry<String, BlobStore>> providers) {
+        this.providers = providers;
```

Comment:
Should this defensively copy, or instead just offer an add and remove interface?